Update docs in order to avoid calling deprecated methods `setTemplate()` and `setTemplates()` in `AbstractAdmin` (#6538)
sonata-project/SonataAdminBundle: docs/reference/templates.rst

Changes made using the ``setTemplate()`` and ``setTemplates()`` methods
override the customizations made in the configuration file, so you can specify

> a global custom template and then override that customization for a specific

Reviewer: Separate commit here too

Author: I agree about the others are grammar fixes, but this is a change about the meaning between the previous and the current behavior.
The previous setting was required **on** a specific admin definition, while the current one is required **for** a specific admin definition.

Reviewer: Oh yes, you are definitely right, I overlooked that one 👍

``Admin`` class.

Finding configured templates
----------------------------
Each ``Admin`` has a ``TemplateRegistry`` service connected to it that holds
the templates registered through the configuration above. The ``TemplateRegistry``
is available through ``$this->getTemplateRegistry()`` within the ``Admin``.

> Using the service container the template registries can be accessed outside an ``Admin``.

Reviewer: That sentence lacks punctuation IMO, also it might be better not to mention the service container other one people might end up using `$container->get` over DI.
```suggestion
Outside an ``Admin``, you can obtain the template registries with their service ID.
```

Use the ``Admin`` code + ``.template_registry`` as the service ID (i.e.

> "app.admin.post" uses the template registry "app.admin.post.template_registry").

Reviewer: I'd use code markup here, but it's funny, [that's the second time I have this discussion today](https://github.com/doctrine/dbal/pull/4384#discussion_r511178109), so don't hesitate to tell me if you feel strongly otherwise.

```suggestion
`app.admin.post` uses the template registry `app.admin.post.template_registry`).
```

Author: I think there is no established rule for the case where we need to refer to service names, parameter names, package names, etc. (since they are **names**, sometimes it feels natural to use double quotes and treat them a literal strings).
Personally, I try to be consistent with this, but in some cases it depends on the context, I think.
In the same reasoning, I try to use backticks when referencing code snippets (with partial or complete fragments, IE `AdminInterface::id()`).

Reviewer: FYI the discussion I linked resolved in using backticks (there was no debate, we needed both backticks _and_ quotes). I tend to use backticks for everything. So to me you can write "admin class", you can also write `Admin` class, but you can't write Admin class, because now you are using a technical name as it appears in the code, and names that appear in code may contain special characters that have meaning in markdown like underscores or stars, but also in normal language, for example dots. When you use backticks, a monospace font will be used, and the reader knows you are talking about something that will appear in the code, and not approximately referring to something using a high level term like "unit of work" or "repository interface".
